sdks/java/core/build.gradle

```diff
@@ -38,6 +38,14 @@ processResources {
   ]
 }
 
+configurations {
+    flinkValidatesRunner
+    sparkValidatesRunner {
```

Comment:
I actually don't understand why this wouldn't be `runners:spark:ValidatesRunner`

Reply:
These aren't tasks but are configurations so the question doesn't make total sense.

If your asking why aren't these defined within `runners/spark/build.gradle`, `runners/flink/build.gradle` then it is because Gradle's Java plugin test task type can only run tests from a directory which would require unzipping the sdk core test jar artifact. Ben and I tried this initially but were unable to find a good way to get the artifact without reaching into the gradle project directly like we do with `sourcesets.test.output` which we want to avoid.

I also do like having these all defined in one place because then a runner just adds a little configuration to the list and it also makes a common place to maintain them all.

Reply:
@lukecwik While this should group validates-runner tests nicely in _most_ cases, Spark appears to group a few of its own tests into the maven validates-runner profile. If we insist on translating all maven validates-runner profiles into gradle and keeping them in one place, then we'll have to reach into another package at some point.

I'm for going the gradle way and splitting tests into the packages that define them. As long as those tests in the spark module run somehow, I don't think it matters that they aren't bundled with the rest of the validates-runner tests.

Reply:
I'm OK with a workaround to get this started, but would keep the goal that we could e.g. delete a runner directory entirely without affecting the core SDK build. So textual references are also a form of dependency I'd prefer to avoid. When the dependency graph of tasks doesn't match the dependency graph of modules, you have some real spaghetti potential.

Reply:
Conceptually, the ValidatesRunner could be its own module depending on the runner and the core SDK, so the choice of which to lump it into is really based on the fact that the runner depends on the core SDK.

Similarly, the integration of a runner and the examples is its own module that depends on both, and the choice to lump it in with the examples is because the examples depend on the runner.

Reply:
What do you mean by "textual references"? This is a real task-level dependency that gradle is aware of. If something breaks this dependency chain, gradle will complain up-front (i.e, at evaluation time). Again, I would argue that setting up the task dependencies in this way is _less_ brittle than setting a hard module evaluation dependency. Especially since in that case we would be reaching in and grabbing/extracting a jar behind gradle's back.

Reply:
I mean that there exists a file under `sdks/java/core` that references a particular runner. That is undesirable. But we should go ahead with this now and figure out how to get gradle to do it right later.

Reply:
OK. I'll leave it as is for now. As a longer-term fix, how do you feel about separating out ValidatesRunner into its own gradle module (and directory) and having that depend on each runner? I wouldn't want to make that change before removing the maven build entirely, but I'm trying to figure out if there's a proper way forward from here.

Reply:
After some additional massaging, I was able to get the tests running from within the Flink directory without manually unzipping jars. It does impose a strict evaluation dependency between modules, but at least it goes in the direction we would prefer. I'll add that as a separate commit on top of this. Let me know if this is satisfactory.

Reply:
Do you have a place to look at the change, either a commit on github or another PR that is not for review to show what you tried?

Reply:
Nevermind, found it here: https://github.com/apache/beam/pull/4436